presto-main/src/main/java/com/facebook/presto/server/WorkerResource.java

```diff
@@ -78,6 +78,33 @@ public Response getStatus(@PathParam("nodeId") String nodeId)
         return Response.ok(responseStream, APPLICATION_JSON_TYPE).build();
     }
 
+    @GET
+    @Path("{nodeId}/thread")
+    public Response getThreads(@PathParam("nodeId") String nodeId)
+    {
+        Set<Node> nodes = nodeManager.getNodes(NodeState.ACTIVE);
```

Comment:
Add a method for this. It can do
```java
.findFirst()
.orElseThrow(() -> new WebApplicationException(NOT_FOUND))
```
You could add this in the initial commit for this class